sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOTest.java

```diff
@@ -629,6 +630,24 @@ public void testBadWildcardRecursive() throws Exception {
     pipeline.apply(TextIO.Read.from("gs://bucket/foo**/baz"));
   }
 
+  /** Options for testing. */
+  public interface RuntimeTestOptions extends PipelineOptions {
+    ValueProvider<String> getInput();
+    void setInput(ValueProvider<String> value);
+
+    ValueProvider<String> getOutput();
+    void setOutput(ValueProvider<String> value);
+  }
+
+  @Test
+  public void testRuntimeOptionsNotCalledInApply() throws Exception {
+    Pipeline pipeline = TestPipeline.create();
+    RuntimeTestOptions options = PipelineOptionsFactory.as(RuntimeTestOptions.class);
+    pipeline
+        .apply(TextIO.Read.from(options.getInput()).withoutValidation())
+        .apply(TextIO.Write.to(options.getOutput()).withoutValidation());
```

Comment:
sorry; can you remind me why these are `withoutValidation`? I'd hope that the validation code, assuming any exists, will see VPs that are not accessible and simply skip validation those options -- but will validate any others.